### DISCUSSION TITLE
datetime and timestamps both in database

### DISCUSSION THREAD

**Discussion by eikaramba** [directus/directus#14481] [Archive]
On multiple system i observed the same "problem" now that i see both timestamps and datetimes for the date_created fields of my collections (set by using the standard fields during create).

![grafik](https://user-images.githubusercontent.com/1320525/179424834-c497393a-6296-4033-bc4c-bd97afd6b048.png)

The problem is that displaying is fine but filtering is not working anymore. Basically both in api and app filtering only affects one part of the values, not the other.
you can also see this here, where i sorted by date_created but actually get a "break" in the middle of the data set
![grafik](https://user-images.githubusercontent.com/1320525/179424889-8b86857d-0a3d-402b-aff8-a2defad76e78.png)

now my problem is
a) i don't know what the correct value is, datetime or timestamp? i am using SQLITE btw and newest directus 9.14.1

b) i don't know exactly how this happens but suspect either that it was after an upgrade&migration or after i might have updated the "on create" behaviour to set it to "nothing" instead of "current date".

What i can say is that i did not change anything else and merely just the system as it is.

**Comment by rijkvanzanten** [comment]
> or after i might have updated the "on create" behaviour to set it to "nothing" instead of "current date".

Does the column in the database have a default value set too? Maybe the default value generated by SQLite is different from the value Directus generates with the "on update" configuration 🤔

**Reply by eikaramba** [reply]
the field does not have a default value set in the database schema. also creating a new entry by hand in the database is not setting it automatically (it is NULL).

**Comment by eikaramba** [comment]
could it be that directus is setting the timestamp as a datetime value and once i switched to on create = "do nothing" and set the date myself that then i got the timestamp values? That is currently what i think is happening. but then it seems the logic is incosistent within directus.

**Reply by rijkvanzanten** [reply]
Could be! Little tricky to say yet

**Reply by eikaramba** [reply]
i can help debugging this once i am back from vacation. basicallly just by replicating the circumstances. one question however: should timestamp be the "correct" value? i see that all fields are normally given "cast-timestamp" as an attribute in the field collections for directus.

**Comment by eikaramba** [comment]
i tried to reproduce this on a "fresh" installation. via rest, via app, with "on create" setting and without... i could not reproduce it. it always saved the dates as timestamps in the database. therefore i suspect it was a migration issue and hopefully will not occur again.
